mono/mini/aot-compiler.c

```diff
@@ -12630,7 +12630,7 @@ mono_compile_assembly (MonoAssembly *ass, guint32 opts, const char *aot_options,
 		}
 	}
 
-	if (mono_aot_mode_is_full (&acfg->aot_opts)) {
+	if (mono_aot_mode_is_full (&acfg->aot_opts) || mono_aot_mode_is_interp (&acfg->aot_opts)) {
 		acfg->flags = (MonoAotFileFlags)(acfg->flags | MONO_AOT_FILE_FLAG_FULL_AOT);
```

Comment:
@vargaz you mentioned on Slack that we shouldn't use this this for `--aot=interp`. 

I'll look into adding `MONO_AOT_FILE_FLAG_INTERP`